tests/test_views.py

```diff
@@ -1281,3 +1316,193 @@ def test_get_course_specific_consent_for_enrolled_user(
+        dummy_demo_course_details_data['effort'] = 'Cannot convert to integer'
+        client.get_course_details.return_value = dummy_demo_course_details_data
+        self._login()
+        enterprise_customer = EnterpriseCustomerFactory(
+            name='Starfleet Academy',
+            enable_data_sharing_consent=True,
+            enforce_data_sharing_consent='at_enrollment',
+        )
+        enterprise_landing_page_url = reverse(
+            'enterprise_course_enrollment_page',
+            args=[enterprise_customer.uuid, course_id],
+        )
+        response = self.client.get(enterprise_landing_page_url)
+        assert response.status_code == 200
+        course_modes = [
+            {
+                'mode': 'enroll',
+                'title': 'Enroll',
+                'original_price': '$200',
+                'final_price': '$190',
+                'description': 'Discount provided by Starfleet Academy',
+            }
+        ]
+        expected_context = {
+            'platform_name': 'edX',
+            'page_title': 'Choose Your Track',
+            'course_id': course_id,
+            'course_name': self.dummy_demo_course_details_data['name'],
+            'course_organization': self.dummy_demo_course_details_data['org'],
+            'course_short_description': self.dummy_demo_course_details_data['short_description'],
+            'course_pacing': self.dummy_demo_course_details_data['pacing'].title(),
+            'course_start_date': parse(self.dummy_demo_course_details_data['start']).strftime('%B %d, %Y'),
+            'course_image_uri': self.dummy_demo_course_details_data['media']['course_image']['uri'],
+            'enterprise_customer': enterprise_customer,
+            'enterprise_welcome_text': (
+                "<strong>Starfleet Academy</strong> has partnered with <strong>edX</strong> to "
+                "offer you high-quality learning opportunities from the world's best universities."
+            ),
+            'confirmation_text': 'Confirm your course',
+            'starts_at_text': 'Starts',
+            'course_pace_text': 'Paced',
+            'view_course_details_text': 'View Course Details',
+            'select_mode_text': 'Please select one:',
+            'price_text': 'Price',
+            'continue_link_text': 'Continue',
+            'course_modes': course_modes,
+            'course_effort': '',
+            'level_text': 'Level',
+            'effort_text': 'Effort',
+            'duration_text': 'Average Length',
+            'course_overview': None,
+            'organization_logo': 'logo.png',
+            'organization_name': 'Organization',
+            'duration': '12 weeks per course',
+            'course_level_type': 'Type 1',
+            'close_modal_button_text': 'Close',
+        }
+        for key, value in expected_context.items():
+            assert response.context[key] == value  # pylint: disable=no-member
```

Comment:
@replaceafill Thanks for adding these tests!

There's a lot of duplication between them. Aside from the method name, the only lines that differ are:

```python
# From test_get_course_enrollment_page_with_no_course_effort:
        dummy_demo_course_details_data['effort'] = None
# From test_get_course_enrollment_page_with_unparseable_course_effort:
        dummy_demo_course_details_data['effort'] = 'Cannot convert to integer'
```

Could you DRY them out using `ddt`?

Also, if you could add some inline comments about what's happening (as well as a bit of vertical whitespace to group logical units), that would make the code easier to digest.